File: script/LocationGenerator.java
import java.util.Random;
public class LocationGenerator {
    /**
     * Randomly generates the address of the user/business
     * @return the elements of a location object (user's/business's address) in an array
     */
    public static String[] GenerateAddress(Random random) {
      //predefined lists for the location components
      String[] streetNames = {"Hillary Cresenct", "Elizabeth Street", "Alice Avenue", "Racheal Road", "Peveral Street", "Moorhouse Avenue", "Riccarton Road", "Clyde Road", "Angelic Avenue", "Henley Road", "Sandspit Road", "Dinglebank Road", "Waipa Street", "Ranui Avenue", "Marywil Crescent"};
      String[] cities = {"Dunedin", "Nightcaps", "Gore", "Tapanui", "Wellington", "Christchurch", "Auckland", "Melbourne", "Brisbance", "Sydeny", "Perth", "Darwin", "Alice Springs", "Tokyo", "London"};
      String[] regions = {"Otago", "Southland", "Canterbury", "Victoria", "Tasman", "Upper Hutt", "Alaska", "California", "Florida", "Hubei Province", "Fujian Province", "Hainan Province", "Munster", "Ulster", "Leinster"};
      String[] countries = {"New Zealand", "Malaysia", "Australia", "England", "United Kingdom", "Japan", "Korea", "Singapore", "France", "Germany", "Norway", "Ireland", "Belgium", "Iceland", "Thailand"};
      //add a choice for the randomizer to select an empty district since district can be empty too
      String[] districts = {"", "Kaipara District", "New Plymouth District", "Carterton District", "Beaver County", "Big Lakes County", "Camrose County", "Cardston County", "Gambir", "Menteng", "Senen", "Johor Bahru District", "Jeli District", "Raub District", "Ruapehu District"};

      String streetNum = String.valueOf(random.nextInt(998) + 1);
      String streetName = streetNames[random.nextInt(streetNames.length)];
      String city = cities[random.nextInt(cities.length)];
      String region = regions[random.nextInt(regions.length)];
      String country = countries[random.nextInt(countries.length)];
      String postcode = String.valueOf(random.nextInt(98999) + 1000);
      String district = districts[random.nextInt(districts.length)];
      String[] address = {streetNum, streetName, city, region, country, postcode, district};
      return address;
  }
}

File: script/location generator/LocationGenerator.py
import csv
import random
#list of country names to be randomized
COUNTRY_LIST = ["New Zealand", "Singapore", "Malaysia", "China", 
                "United States of America", "Sweden", "Norway", "Indonesia",
                "Canada", "Japan", "Korea", "Iceland", "Kenya", "Thailand", 
                "Ireland", "Belgium", "Italy", "Argentina", "Brazil",
                "Australia"]
#list of state names around the world to be randomized
STATE_LIST = ["Alaska", "California", "Florida", "Canterbury", "Auckland",
              "Bay of Plenty", "Hubei Province", "Fujian Province",
              "Hainan Province", "Munster", "Ulster", "Leinster", "Bangkok",
              "Krabi", "Phuket", "Tokyo", "Kyoto", "Okinawa", "Nelson", "Otago"]
#list of district names around the world to be randomized
DISTRICT_LIST = ["", "Kaipara District", "New Plymouth District",
                 "Carterton District", "Beaver County", "Big Lakes County",
                 "Camrose County", "Cardston County", "Gambir", "Menteng",
                 "Senen", "Johor Bahru District", "Jeli District", 
                 "Raub District", "Ruapehu District", "South Wairarapa District",
                 "Hastings District", "Manawatu District", "Southland District",
                 "Tasman District"]

with open('nz-street-address.csv', encoding="utf8") as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    chosen_row = random.choice(list(csv_reader))
    print(", ".join(chosen_row))
    #line_count = 0
    #for row in csv_reader:
        #if line_count == 0:
            #print(f'Column names are {", ".join(row)}')
            #line_count += 1
            #break;
        #else:
            #print(f'\t{row[0]} works in the {row[1]} department, and was born in {row[2]}.')
            #line_count += 1
    #print(f'Processed {line_count} lines.')